pulp_smash/pulp3/pytest_plugin/__init__.py

```diff
@@ -0,0 +1,236 @@
+
+@pytest.fixture
+def gen_fixture_server(gen_threaded_aiohttp_server):
+    def _gen_fixture_server(fixtures_root, ssl_ctx):
+        app = web.Application()
+        call_record = add_recording_route(app, fixtures_root)
+        return gen_threaded_aiohttp_server(app, ssl_ctx, call_record)
+
+    yield _gen_fixture_server
+
+
+@pytest.fixture(scope="session")
+def pulpcore_client():
+    configuration = cfg.get_bindings_config()
+    return ApiClient(configuration)
+
+
+@pytest.fixture(scope="session")
+def tasks_api_client(pulpcore_client):
+    return TasksApi(pulpcore_client)
+
+
+## Oprhan Handling Fixtures
+
+
+@pytest.fixture
+def delete_orphans_pre():
+    delete_orphans()
+    yield
+
+
+## Server Side TLS Fixtures
+
+
+@pytest.fixture
+def tls_certificate_authority():
+    return trustme.CA()
+
+
+@pytest.fixture
+def tls_certificate_authority_cert(tls_certificate_authority):
+    return tls_certificate_authority.cert_pem.bytes().decode()
+
+
+@pytest.fixture
+def tls_certificate(tls_certificate_authority):
+    return tls_certificate_authority.issue_cert(
+        cfg.aiohttp_fixtures_origin,
+    )
+
+
+## Client Side TLS Fixtures
+
+
+@pytest.fixture
+def client_tls_certificate_authority():
+    return trustme.CA()
+
+
+@pytest.fixture
+def client_tls_certificate_authority_pem_path(client_tls_certificate_authority):
+    with client_tls_certificate_authority.cert_pem.tempfile() as client_ca_pem:
+        yield client_ca_pem
+
+
+@pytest.fixture
+def client_tls_certificate(client_tls_certificate_authority):
+    return client_tls_certificate_authority.issue_cert(
+        cfg.aiohttp_fixtures_origin,
+    )
+
+
+@pytest.fixture
+def client_tls_certificate_cert_pem(client_tls_certificate):
+    return client_tls_certificate.cert_chain_pems[0].bytes().decode()
+
+
+@pytest.fixture
+def client_tls_certificate_key_pem(client_tls_certificate):
+    return client_tls_certificate.private_key_pem.bytes().decode()
+
+
+## SSL Context Fixtures
+
+
+@pytest.fixture
+def ssl_ctx(tls_certificate):
+    ssl_ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    tls_certificate.configure_cert(ssl_ctx)
+    return ssl_ctx
+
+
+@pytest.fixture
+def ssl_ctx_req_client_auth(
+    tls_certificate, client_tls_certificate, client_tls_certificate_authority_pem_path
+):
+    ssl_ctx = ssl.create_default_context(
+        purpose=ssl.Purpose.CLIENT_AUTH, cafile=client_tls_certificate_authority_pem_path
+    )
+    ssl_ctx.verify_mode = ssl.CERT_REQUIRED
+    tls_certificate.configure_cert(ssl_ctx)
+    return ssl_ctx
+
+
+## Object Cleanup fixtures
+
+
+@pytest.fixture
+def gen_object_with_cleanup():
+    obj_hrefs = []
+
+    def _gen_object_with_cleanup(api_client, data):
+        new_obj = api_client.create(data)
+        obj_hrefs.append((api_client, new_obj.pulp_href))
+        return new_obj
+
+    yield _gen_object_with_cleanup
+
+    delete_task_hrefs = []
+    for api_client, pulp_href in obj_hrefs:
+        task_url = api_client.delete(pulp_href).task
```

Comment:
I'm not sure we can rely on all models to only delete in tasks.

Reply:
I agree, but we can use this for all of those types of models. I was thinking the name should reflect this better, but I didn't come up with anything.

Reply:
Agreed to wait for this to become a problem.
(autodetect is still my preferred solution)